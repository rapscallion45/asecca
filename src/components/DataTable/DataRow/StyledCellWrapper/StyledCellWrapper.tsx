import { styled } from '@mui/material/styles';
import { TableCell } from '@mui/material';
import { tableCellClasses } from '@mui/material/TableCell';

/**
 * Styled Table Cell Wrapper
 *
 * Wrapper for application specific styling of a data table cell
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @component
 * @returns {Component} - styled table data cell wrapper
 */
const StyledTableCell = styled(TableCell)(({ theme }) => ({
  /* table head colors */
  [`&.${tableCellClasses.head}`]: {
    backgroundColor: theme.palette.common.black,
    color: theme.palette.common.white,
  },
  /* adjust font size */
  [`&.${tableCellClasses.body}`]: {
    fontSize: 14,
  },
}));

export default StyledTableCell;
