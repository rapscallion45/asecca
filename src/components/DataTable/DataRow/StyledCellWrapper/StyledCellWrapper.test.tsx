import React from 'react';
import renderer from 'react-test-renderer';
import StyledCellWrapper from './StyledCellWrapper';

/**
 * Styled Table Cell Wrapper Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 */
describe('StyledCellWrapper', () => {
  it('Renders correctly', async () => {
    /** perform snapshot test */
    const tree = renderer.create(<StyledCellWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
