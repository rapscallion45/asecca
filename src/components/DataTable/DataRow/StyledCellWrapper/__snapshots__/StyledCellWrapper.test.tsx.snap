// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledCellWrapper Renders correctly 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-sizeMedium css-1xspv29-MuiTableCell-root"
/>
`;
